### ISSUE TITLE
GLTFLoader: No texture.source should be valid?

### ISSUE THREAD

**Issue by takahirox** [mrdoob/three.js#24846]
**Describe the bug**

I failed to load a glTF asset 

1. that has `texture` that doesn't have `source` but has `extensions.foo_extension.source`, like the following example, and 
2. that doesn't have `foo_extension` in `extensionsRequired`, and
3. a plugin for handling `foo_extension` is not registered to glTF loader

```
textures: [
  {
    "extensions": {
      "foo_extension": {
        source: 0,
        ...
      }
    }
  }
]
```

The error log is

> Cannot read properties of undefined (reading 'uri')

`GLTFParser.loadTexture()` seems to handle `texture` as if `texture.source` is required.

https://github.com/mrdoob/three.js/blob/321106c2bfd447ccfd5fb2a077af366955733291/examples/jsm/loaders/GLTFLoader.js#L3152-L3160

**To Reproduce**

T.B.D.

***Live example***

T.B.D.

**Expected behavior**

Perhaps the loader shouldn't fail to load it? Because in the glTF spec

1. `texture.source` is not required

https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#reference-texture

2. `texture.source` description is

> The index of the image used by this texture. When undefined, an extension or other mechanism SHOULD supply an alternate texture source, otherwise behavior is undefined.

in the example, `texture.extensions.foo_extension` supplies a texture source (but the extension is unknown by the glTF loader).

3. 3.8.2 Texture section describs that

https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#textures

> When texture.source is undefined, the image SHOULD be provided by an extension or application-specific means, otherwise the texture object is undefined.
> Implementation Note
> Client implementations may render such textures with a predefined placeholder image or being filled with some error color (usually magenta).

If we allow no `texture.source`, `GLTFParser.loadTexture()` will return `null`?

**Alternate solutions**

If we don't allow undefined behavior texture object, the glTF asset should have `foo_extension` in `extensionsRequired`?

The glTF spec mentions that

> All glTF extensions required to load and/or render an asset MUST be listed in the top-level extensionsRequired array

https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#specifying-extensions

**Platform:**

 - Device: Desktop but probably any
 - OS: Windows but probably any
 - Browser: Chrome but probably any
 - Three.js version: dev

/cc @donmccurdy

**Comment by donmccurdy** [comment]
> The index of the image used by this texture. When undefined, an extension or other mechanism SHOULD supply an alternate texture source, otherwise behavior is undefined.

How I read this would be that the file is not invalid under the core glTF specification, but when `.source` is omitted, it's very much the expectation that _something_ will be providing that texture's data. Generally that's the extension, and the core specification is just leaving some flexibility for extensions to define implementation details.  With `KHR_texture_basisu` (the only ratified Khronos Group extension currently extending textures) you'll note that the extension itself [specifies](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_texture_basisu/README.md#using-without-a-fallback):

> To use KTX v2 image with Basis Universal supercompression without a fallback, define KHR_texture_basisu in both extensionsUsed and extensionsRequired. The texture object will then have its source property omitted as shown below.

In other words, `KHR_texture_basisu` must be required if assigned to a texture whose base `.source` is undefined. 

***

tl;dr — I think it's OK for us to throw an error here, or at least the specification doesn't say otherwise. If we think it would be better to something else (skip the texture? return a bright purple placeholder texture?) then we could certainly discuss that.

**Comment by takahirox** [comment]
Thanks for the response.

Yeah, it's the best if extension specifications defines whether the extensions must be marked as required.

Let me keep this issue opened for a week so far because someone else might have different opinions.

**Comment by takahirox** [comment]
No other comments. That extension should be marked as required in case of no texture.source in the core makes sense to me. Closing.
